src/components/matrix/column-headers.tsx

```diff
@@ -0,0 +1,98 @@
+import { memo, forwardRef, ForwardedRef, CSSProperties } from 'react';
+import { VariableSizeList as List, ListOnScrollProps } from 'react-window';
+
+type ColumnHeadersProps = {
+  cellHeight?: number; // height of grid cell
+  cellWidth: number; // width of grid cell
+  height: number; // height of each column header
+  width: number; // width of all column headers
+  itemCount: number; // number of items
+  itemData?: any; // data passed to each column
+  left: number; // how much to position the column left by
+  onScroll?: ((props: ListOnScrollProps) => any) | undefined;
+};
+
+/**
+ * Virtualized Column Header that displays headers as they scroll into the given width
+ */
+const ColumnHeaders = (
+  { left, cellWidth, height, width, itemCount, itemData, onScroll }: ColumnHeadersProps,
+  ref: ForwardedRef<any>
+): JSX.Element => {
+  const { listData } = itemData;
+
+  /**
+   * Gets item size based on given index
+   * Uses a different item size for the last margin column
+   *
+   * @param {number} index
+   * @returns {number} item size
+   */
+  const itemSize = (index: number) => (index < listData[0].length - 1 ? cellWidth : cellWidth + 30);
+
+  const columnHeadersStyles: CSSProperties = {
+    position: 'absolute',
+    left: left,
+  };
+
+  type HeaderComponentProps = {
+    index: number;
+    data: any;
+    style: CSSProperties;
+  };
+
+  /**
+   * Header component for each column
+   */
+  const HeaderComponent = ({ index, data, style }: HeaderComponentProps): JSX.Element => {
```

Comment:
I'm not familiar with how this works. Why is this component defined inside the other component? If it doesn't have to be, move it to be outside so it's easier to read.

Reply:
moved outside